testsuite/MDAnalysisTests/coordinates/test_dcd.py

```diff
@@ -65,420 +82,287 @@ def test_with_statement(self):
             assert_almost_equal(ts_orig.dimensions, ts_copy.dimensions, 4,
                                 err_msg="DCD->DCD: unit cell dimensions wrong "
                                 "at frame {0}".format(ts_orig.frame))
-        del w
-
-
-class TestDCDReader_CHARMM_Unitcell(_TestDCDReader_TriclinicUnitcell,
-                                    RefCHARMMtriclinicDCD):
-    __test__ = True
 
 
-class TestDCDReader_NAMD_Unitcell(_TestDCDReader_TriclinicUnitcell,
-                                  RefNAMDtriclinicDCD):
-    __test__ = True
-
-
-class TestNCDF2DCD(TestCase):
-    @dec.skipif(module_not_found("netCDF4"),
-                "Test skipped because netCDF is not available.")
-    def setUp(self):
-        self.u = mda.Universe(PRMncdf, NCDF)
-        # create the DCD
-        self.tmpdir = tempdir.TempDir()
-        self.dcd = self.tmpdir.name + '/ncdf-2-dcd.dcd'
-        DCD = mda.Writer(self.dcd, n_atoms=self.u.atoms.n_atoms)
-        for ts in self.u.trajectory:
-            DCD.write(ts)
-        DCD.close()
-        self.w = mda.Universe(PRMncdf, self.dcd)
-
-    def tearDown(self):
-        try:
-            os.unlink(self.dcd)
-        except (AttributeError, OSError):
-            pass
-        del self.u
-        del self.w
-        del self.tmpdir
-
-    @attr('issue')
-    def test_unitcell(self):
-        """NCDFReader: Test that DCDWriter correctly writes the CHARMM
-        unit cell"""
-        for ts_orig, ts_copy in zip(self.u.trajectory,
-                                    self.w.trajectory):
-            assert_almost_equal(
-                ts_orig.dimensions,
-                ts_copy.dimensions,
-                3,
-                err_msg="NCDF->DCD: unit cell dimensions wrong at frame {0:d}".format(
-                ts_orig.frame))
-
-    def test_coordinates(self):
-        for ts_orig, ts_copy in zip(self.u.trajectory,
-                                    self.w.trajectory):
-            assert_almost_equal(
-                self.u.atoms.positions,
-                self.w.atoms.positions,
-                3,
-                err_msg="NCDF->DCD: coordinates wrong at frame {0:d}".format(
-                ts_orig.frame))
-
-
-class TestDCDTimestep(BaseTimestepTest):
-    Timestep = mda.coordinates.DCD.Timestep
-    name = "DCD"
-    has_box = True
-    set_box = True
-    unitcell = np.array([10., 90., 11., 90., 90., 12.])
-    uni_args = (PSF, DCD)
-
-    def test_ts_order_define(self):
-        """Check that users can hack in a custom unitcell order"""
-        old = self.Timestep._ts_order
-        self.ts._ts_order = [0, 2, 5, 1, 3, 4]
-        self.ts.dimensions = np.array([10, 11, 12, 80, 85, 90])
-        assert_allclose(self.ts._unitcell, np.array([10, 80, 11, 85, 90, 12]))
-        self.ts._ts_order = old
-        self.ts.dimensions = np.zeros(6)
+@pytest.fixture(scope='module')
+def ncdf2dcd(tmpdir_factory):
+    pytest.importorskip("netCDF4")
```

Comment:
OK so this is a funny new thing. After some searching I noticed that `importorskip` isn't a decorator. Rather it can be used as a function in a test to deactivate it. The effect is has in the fixture here is that every test that uses the fixture is automatically skipped when it's run. This is kind of cool since the tests themselves don't need to take care if `netCDF4` is installed now.